pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -229,9 +229,10 @@ func (jm *ControllerV2) sync(ctx context.Context, cronJobKey string) (*time.Dura
 func (jm *ControllerV2) resolveControllerRef(namespace string, controllerRef *metav1.OwnerReference) *batchv1.CronJob {
 	// We can't look up by UID, so look up by Name and then verify UID.
 	// Don't even try to look up by Name if it's the wrong Kind.
-	if controllerRef.Kind != controllerKind.Kind {
+	if controllerRef.Kind != controllerKind.Kind && controllerRef.Kind != "CronJob" {
```

Comment:
This is testing the same thing (see `controllerKind`). But we could add a check for the group (`batch`)